### ISSUE TITLE
Case sensitive font names in unused-fontface-remover.js

### ISSUE THREAD

**Issue by mklepaczewski** [pocketjoso/penthouse#294]
According to the [spec](https://drafts.csswg.org/css-fonts/#localized-name-matching), font-family names are case insensitive. However, in `unused-fontface-remover.js` in `getAllFontNameValues()` font names are extracted to the `fontNameValues variable` - which is a `Set`. This set is then returned to `unusedFontfaceRemover()` and then used in case sensitive manner:

```
const fontNameValues = getAllFontNameValues(ast); 
....
const familyName = decodeFontName(declaration.value); // was this @font-face used?
if (!fontNameValues.has(familyName)) {
  debuglog('drop unused @font-face: ' + familyName);
  used = false;
}
```

As a result, font-family rules that refer to the font using different casing are recognized as not being used.

**Comment by pocketjoso** [comment]
I see how this could appear to be the source of the issue, but the `csstree` parser normalises font-family names in lower case, so this check is already case insensitive. I just tested to confirm.

If you have a bug that can be reproduced, could you share a (minimal) reproducible test case? There must be something else going on.

**Comment by mklepaczewski** [comment]
I'll dig deeper into it and I'll provide minimal test - it may take a while though due to my workload. I'm using penthouse through critical, maybe that's why I see non-normalized font names. FWIW this is a part of the log from my run. It suggests that in my case font names are not converted to lower case (note that the font names are stored as `familyName` variable, which is used against the `fontNameValues`):

```
2020-03-16T15:27:48.703Z penthouse:css-cleanup:unused-font-face-remover found used font-family: Shopkeeper-Icon-Font
2020-03-16T15:27:48.708Z penthouse:css-cleanup:unused-font-face-remover found used font-family: neueeinstellung
2020-03-16T15:27:48.708Z penthouse:css-cleanup:unused-font-face-remover found used font-family: -apple-system
2020-03-16T15:27:48.708Z penthouse:css-cleanup:unused-font-face-remover found used font-family: BlinkMacSystemFont
....
2020-03-16T15:27:48.716Z penthouse:css-cleanup:unused-font-face-remover drop unused @font-face: WooCommerce
2020-03-16T15:27:48.716Z penthouse:css-cleanup:unused-font-face-remover drop unused @font-face: fontawesomesatt
2020-03-16T15:27:48.716Z penthouse:css-cleanup:unused-font-face-remover drop unused @font-face: star
```

**Comment by pocketjoso** [comment]
Interesting, for me all font names are normalised, regardless of where it was declared. I suspect this could be an OS issue. I'm testing from MacOS, where it normalises as it should, and I can confirm that it's the same on Linux.

What OS are you running on?

**Comment by mklepaczewski** [comment]
Windows, I haven't tested it on Linux yet.

**Comment by pocketjoso** [comment]
Would it be possible for you to test on Linux @mklepaczewski ? To see if we can increase confidence that this is OS related.

**Comment by mklepaczewski** [comment]
@pocketjoso Yes, I'll try to do it sometime next week.

**Comment by pocketjoso** [comment]
Do you have any updates on this, @mklepaczewski ?

**Comment by mklepaczewski** [comment]
In progress, eta: a day or two.

**Comment by mklepaczewski** [comment]
Minimal test case: https://github.com/mklepaczewski/penthouse-fontname-bug

Tested both on Ubuntu 18.04.1 and Windows 10. 

Expected output
```
$ node test.js
@font-face{font-family:'Roboto';font-style:normal;font-weight:300;src:local('Roboto Light'),local('Roboto-Light'),url(https://fonts.gstatic.com/s/roboto/v20/KFOlCnqEu92Fr1MmSU5fBBc4.woff2) format('woff2');unicode-range:U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD}#one{color:red;font-family:'Roboto';font-weight:300}#two{color:blue}
```
Actual output
```
$ node test.js
#one{color:red;font-family:'roboto';font-weight:300}#two{color:blue}
```

Note that changing 'roboto' to 'Roboto' in `style.css` `#one` rule produces expected output.

**Comment by pocketjoso** [comment]
Thanks for the test case! I released a fix now, published in [penthouse@2.3.2](https://github.com/pocketjoso/penthouse/releases/tag/v2.3.2).
